src/Hazelcast.Net/Sql/HLocalDate.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Hazelcast.Sql
+{
+    /// <summary>
+    /// Represents Hazelcast SQL <c>DATE</c> type corresponding to <c>java.time.LocalDate</c> in Java.
+    /// </summary>
+    public readonly struct HLocalDate : IEquatable<HLocalDate>
```

Comment:
@zpqrtbnk 
Added deserialization for date/time types to be returned from SQL queries. Now, regarding serialization - to pass them as query parameters...

For now parameters use default serializers (built-in and user-provided), so simplest approach would be to register `IStreamSerializer`s for new types in `ConstantSerializerDefinitions`. However I'm not sure if that's reasonable as I haven't encountered them being used outside of our new SQL feature.

Another question that follows is regarding built-in serialization for `DateTime` and `DateTimeOffset` (plus `DateOnly`, `TimeOnly` in later versions):
- users would definitely like this to be supported out-of-the-box;
- `HLocalDateTime`, `HOffsetDateTime` (`HLocalDate`, `HLocalTime`) contains all allowed range of values for `DateTime`, `DateTimeOffset` (`DateOnly`, `TimeOnly`).

As such, it probably makes sense to support it and just forward call to corresponding `H*` serializer. But before doing this, I want to ensure that there are no other places where date/time types are used yet serialized in a different way.

Reply:
I do believe that date/time types should be supported "by default" for everything, for instance also in maps, not purely in SQL, so I *think* it's reasonable?

Reply:
Agreed with @mdumandag  on adding default serializers for them.